orderer/consensus/smartbft/chain.go

```diff
@@ -196,22 +196,32 @@ func NewChain(
 	heartbeatTicker := time.NewTicker(1 * time.Second)
 	heartbeatTimeout := 10 * time.Second
 	heartbeatCount := uint64(5)
-	var role Role
-	if selfID%2 == 0 {
-		role = HeartbeatSender
-	} else {
-		role = HeartbeatReceiver
-	}
-	senders := make([]uint64, 0)
-	receivers := make([]uint64, 0)
+
+	myRole := HeartbeatSender
+
+	heartbeatReceivers := make([]uint64, 0)
 	for _, node := range currentCommittee {
-		if node.ID%2 == 0 {
-			senders = append(senders, uint64(node.ID))
-		} else {
-			receivers = append(receivers, uint64(node.ID))
+		if selfID == uint64(node.ID) {
+			myRole = HeartbeatReceiver
+		}
+		heartbeatReceivers = append(heartbeatReceivers, uint64(node.ID))
+	}
+
+	heartbeatSenders := make([]uint64, 0)
```

Comment:
please extract this block to a function that returns the senders and the receivers (and your role)